hyperspy/axes.py

```diff
@@ -729,18 +729,80 @@ def update_from(self, axis, attributes):
             any_changes = True
         return any_changes
 
-    def convert_to_uniform_axis(self):
-        """Convert to an uniform axis."""
-        scale = (self.high_value - self.low_value) / self.size
+    def convert_to_uniform_axis(self, keep_bounds=True, log_scale_error=True):
+        """
+        Convert to an uniform axis.
+
+        Parameters
+        ----------
+        keep_bounds : bool
+            If ``True``, the first and last value of the axis will not changed.
+            The new scale is estimated by averaging the :func:`numpy.diff` of
+            the axis. If ``False``, the scale and offset are calculated using
+            :meth:`numpy.polynomial.polynomial.Polynomial.fit`, which minimises
+            the scale difference over the whole axis range but the bounds of
+            the axis can change. Default is ``True``.
+        log_scale_error : bool
+            If ``True``, the maximum scale error will be logged as INFO.
+            Default is ``True``.
+
+        Examples
+        --------
+        Using ``keep_bounds=True`` (default):
+
+        >>> s = hs.data.luminescence_signal(uniform=False)
+        >>> print(s.axes_manager)
+        <Axes manager, axes: (|1024)>
+                    Name |   size |  index |  offset |   scale |  units
+        ================ | ====== | ====== | ======= | ======= | ======
+        ---------------- | ------ | ------ | ------- | ------- | ------
+                  Energy |   1024 |      0 | non-uniform axis |     eV
+        >>> s.axes_manager[-1].convert_to_uniform_axis(keep_bounds=True)
+        >>> print(s.axes_manager)
+        <Axes manager, axes: (|1024)>
+                    Name |   size |  index |  offset |   scale |  units
+        ================ | ====== | ====== | ======= | ======= | ======
+        ---------------- | ------ | ------ | ------- | ------- | ------
+                  Energy |   1024 |      0 |     1.6 |  0.0039 |     eV
+
+        Using ``keep_bounds=False``:
+
+        >>> s = hs.data.luminescence_signal(uniform=False)
+        >>> print(s.axes_manager)
+        <Axes manager, axes: (|1024)>
+                    Name |   size |  index |  offset |   scale |  units
+        ================ | ====== | ====== | ======= | ======= | ======
+        ---------------- | ------ | ------ | ------- | ------- | ------
+                  Energy |   1024 |      0 | non-uniform axis |     eV
+        >>> s.axes_manager[-1].convert_to_uniform_axis(keep_bounds=False)
+        >>> print(s.axes_manager)
+        <Axes manager, axes: (|1024)>
+                    Name |   size |  index |  offset |   scale |  units
+        ================ | ====== | ====== | ======= | ======= | ======
+        ---------------- | ------ | ------ | ------- | ------- | ------
+                  Energy |   1024 |      0 |     1.1 |  0.0033 |     eV
+
+        """
+        indices = np.arange(self.size)
+        if keep_bounds:
+            scale = np.mean(np.diff(self.axis))
```

Comment:
Mathematically equivalent to 
`scale = (self.axis[-1] - self.axis[0])/len(self.axis)`

which is cheaper computationally, and (to me) shows more explicitly that bounds are preserved. What you think?

Reply:
Done!

Reply:
Nice! Also nice that you realized I had off-by-one error ^^